src/class/ResponseHeader/ResponseHeader.cpp

```diff
@@ -0,0 +1,224 @@
+
+#include <ResponseHeader/ResponseHeader.hpp>
+
+/*
+** ------------------------------- CONSTRUCTOR --------------------------------
+*/
+
+ResponseHeader::ResponseHeader(int code, const ErrorPage &default_page)
+{
+	defaultConstructor();
+	setStatusCode(code);
+	if (code >= 400)
+	{
+		setContentType(".html");
+		setConnection("close");
+		defaultErrorPage(code, default_page);
+		if (code == 405)
+			_header[CONNECTION].name = "Allow: ";
+	}
+}
+
+ResponseHeader::ResponseHeader(const ResponseHeader &src)
+	: _header(src.getHeader())
+{
+}
+
+const char *ResponseHeader::InvalidDefaultPage::what() const throw()
+{
+	return "Invalid default page";
+}
+
+/*
+** -------------------------------- DESTRUCTOR --------------------------------
+*/
+
+ResponseHeader::~ResponseHeader()
+{
+}
+
+/*
+** --------------------------------- OVERLOAD ---------------------------------
+*/
+
+ResponseHeader &ResponseHeader::operator=(ResponseHeader const &rhs)
+{
+	if (this != &rhs)
+	{
+		this->_header = rhs.getHeader();
+	}
+	return *this;
+}
+
+std::ostream &operator<<(std::ostream &o, ResponseHeader const &i)
+{
+	o << i.getResponse();
+	return o;
+}
+
+/*
+** --------------------------------- METHODS ----------------------------------
+*/
+
+void ResponseHeader::defaultConstructor()
+{
+
+	for (field f = STANDARD; f <= BODY; f = static_cast<field>(static_cast<int>(f) + 1))
+	{
+		HeaderFields defaultHeader;
+		_header.insert(std::make_pair(f, defaultHeader));
+	}
+	_header[STANDARD].name = "HTTP/1.1 ";
+	_header[STANDARD].value = "";
+	_header[SERVER].name = "Server: ";
+	_header[SERVER].value = "webserv";
+	_header[DATE].name = "Date: ";
+	_header[DATE].value = getCurrentDate();
+	_header[CONNECTION_TYPE].name = "Content-Type: ";
+	_header[CONNECTION_TYPE].value = "";
+	_header[CONTENT_LENGHT].name = "Content-Length: ";
+	_header[CONTENT_LENGHT].value = "";
+	_header[CONNECTION].name = "Connection: ";
+	_header[BODY].name = "\r\n";
+	_header[BODY].value = "";
+	setContentType(".txt");
+	setContentLength(0);
+	setConnection("keep-alive");
+}
+
+void ResponseHeader::defaultErrorPage(int code, const ErrorPage &default_page)
+{
+	HTTPStatusCode default_code = default_page.getStatus();
+	if (code == default_code.get())
+	{
+		std::string path = default_page.getPath().get();
+		std::ifstream file(path.c_str());
+		if (!file)
+			throw ResponseHeader::InvalidDefaultPage();
+		std::stringstream contentStream;
+		contentStream << file.rdbuf();
+		file.close();
+		std::string content = contentStream.str();
+		setBody(content);
+	}
+	else
```

Comment:
this `else` is in case of no error page found?
i think we will need to provide our own default error page

Reply:
This "else" is used when an error page is not defined in the config, I set up our own default error page,  and also is used to set a default body for correct pages.
If it is correct and later they need to change their body, they can